pkg/client/clientcmd/client_config.go

```diff
@@ -109,13 +109,13 @@ func (config DirectClientConfig) ClientConfig() (*client.Config, error) {
 		if err != nil {
 			return nil, err
 		}
-		mergo.Merge(clientConfig, userAuthPartialConfig)
+		mergo.MergeWithOverwrite(clientConfig, userAuthPartialConfig)
```

Comment:
You'll need to update the comments a couple lines up.  They refer to the old merge semantics.